test/api/qcFlags.test.js

```diff
@@ -0,0 +1,218 @@
+                createdById: 2,
+                createdBy: { id: 2, externalId: 456, name: 'Jan Jansen' },
+                flagTypeId: 13,
+                flagType: { id: 13, name: 'Bad', method: 'Bad', bad: true, archived: false, color: null },
+            });
+        });
+        it('should successfuly filter on ids', async () => {
+            const response = await request(server).get('/api/qcFlags?filter[ids][]=1');
+            expect(response.status).to.be.equal(200);
+            const { data, meta } = response.body;
+            expect(meta).to.be.eql({ page: { totalCount: 1, pageCount: 1 } });
+            expect(data).to.be.an('array');
+            expect(data).to.be.lengthOf(1);
+            expect(data[0].id).to.be.equal(1);
+        });
+        it('should successfuly filter on dataPassIds, runNumbers, dplDetectorIds', async () => {
+            const response = await request(server)
+                .get('/api/qcFlags?filter[dataPassIds][]=1&filter[runNumbers][]=106&filter[dplDetectorIds][]=1');
+            expect(response.status).to.be.equal(200);
+            const { data, meta } = response.body;
+            expect(meta).to.be.eql({ page: { totalCount: 3, pageCount: 1 } });
+            expect(data).to.be.an('array');
+            expect(data).to.be.lengthOf(3);
+            expect(data.map(({ id }) => id)).to.have.all.members([1, 2, 3]);
+        });
+
+        it('should successfuly filter on simulationPassIds, runNumbers, dplDetectorIds', async () => {
+            const response = await request(server)
+                .get('/api/qcFlags?filter[simulationPassIds][]=1&filter[runNumbers][]=106&filter[dplDetectorIds][]=1');
+            expect(response.status).to.be.equal(200);
+            const { data, meta } = response.body;
+            expect(meta).to.be.eql({ page: { totalCount: 1, pageCount: 1 } });
+            expect(data).to.be.an('array');
+            expect(data).to.be.lengthOf(1);
+            expect(data[0].id).to.be.equal(5);
+        });
+
+        it('should retrive no records when filtering on ids', async () => {
+            const response = await request(server).get('/api/qcFlags?filter[ids][]=9999');
+            expect(response.status).to.be.equal(200);
+            const { data } = response.body;
+            expect(data).to.be.an('array');
+            expect(data).to.be.lengthOf(0);
+        });
+        it('should successfuly filter on createdBy', async () => {
+            const response = await request(server).get('/api/qcFlags?filter[createdBy][]=John%20Doe');
+            expect(response.status).to.be.equal(200);
+            const { data } = response.body;
+            expect(data).to.be.an('array');
+            expect(data).to.be.lengthOf(3);
+            expect(data.map(({ id }) => id)).to.have.all.members([1, 2, 3]);
+        });
+        it('should support sorting by id', async () => {
+            const response = await request(server).get('/api/qcFlags?sort[id]=ASC');
```

Comment:
If you test sorting, you should test both ways. 
Or one way here but both ways in the service?

Reply:
tests added